tox.ini

```diff
@@ -1,8 +1,80 @@
+# Generated from:
+# https://github.com/zopefoundation/meta/tree/master/config/pure-python
 [tox]
-envlist = py27,py35,py36,py37,py38,pypy, pypy3
+minversion = 3.18
+envlist =
+    lint
+    py27
+    py35
+    py36
+    py37
+    py38
+    py39
+    pypy
+    pypy3
+    docs
+    coverage
 
 [testenv]
+usedevelop = true
+deps =
+    zope.testrunner
+commands =
+    zope-testrunner --test-path=src {posargs:-vc}
+extras =
+    test
+
+[testenv:lint]
+basepython = python3
+skip_install = true
+deps =
+    flake8
+    check-manifest
+    check-python-versions
+commands =
+    flake8 src setup.py
+    check-manifest
+    check-python-versions
+
+[testenv:docs]
+basepython = python3
+skip_install = false
+deps =
+extras =
+    docs
+commands_pre =
 commands =
-    zope-testrunner --test-path=src []
+    sphinx-build -b html -d docs/_build/doctrees docs docs/_build/html
+
+[testenv:coverage]
+basepython = python3
+allowlist_externals =
+    mkdir
 deps =
-    .[test]
+    coverage
+    coverage-python-version
+    zope.testrunner
+commands =
+    mkdir -p {toxinidir}/parts/htmlcov
+    coverage run -m zope.testrunner --test-path=src {posargs:-vc}
+    coverage html
+    coverage report -m --fail-under=99
+
+[coverage:run]
+branch = True
```

Comment:
Is the coverage fail caused by using branch coverage now?

Reply:
I think so, because according to https://github.com/zopefoundation/zope.app.tree/pull/9/checks?check_run_id=2166307604 we have no uncovered lines but just 7 code branches which are only partly tested.